tests/odbc/test-odbc-mssql.cpp

```diff
@@ -75,6 +75,34 @@ TEST_CASE("MS SQL long string", "[odbc][mssql][long]")
     );
 }
 
+TEST_CASE("MS SQL unicode string", "[odbc][mssql][unicode]")
+{
+    soci::session sql(backEnd, connectString);
+
+    struct unicode_table_creator : public table_creator_base
+    {
+        explicit unicode_table_creator(soci::session& sql)
+            : table_creator_base(sql)
+        {
+            sql << "create table soci_test ("
+                    "str nvarchar(255) null"
+                ")";
+        }
+    } unicode_table_creator(sql);
+
+    // std::string str_in { "สวัสดีชาวโลก!" };
+    std::string str_in { u8"\u0E2A\u0E27\u0E31\u0E2A\u0E14\u0E35\u0E0A\u0E32\u0E27\u0E42\u0E25\u0E01!" };
```

Comment:
I don't think we want to use `u8` here, this makes the literal `char8_t` in C++20 and so this won't compile (because `std::string` expects `char`).